test/runnable/minimal2.d

```diff
@@ -30,7 +30,8 @@ void poorMansAssert(bool condition)
 {
     if (!condition)
     {
-        asm {hlt;}
+        static char* hlt;
+        *hlt = 0;
```

Comment:
Newer compilers can use `-checkaction=halt` which will cause `assert(0)` to emit a `UD2` instruction, which is presumably better than a null pointer seg fault.
Or you can use:
```
*cast(byte*)null = 0;
```
which I use from time to time.

Reply:
I thought that the backend errors about detected null pointer dereferences.

Reply:
Well this is all green so it appears not in this case.

Reply:
I meant in response to @WalterBright's suggestion.
```
$ cat halt.d 
void main(string[] args)
{
    *cast(byte*)null = 0;
}
$ dmd halt.d -O
Error: null dereference in function _Dmain
```